### PR TITLE
chore: clippy & fix

### PR DESCRIPTION
**Pull request by ChengYueJia**
* cargo clippy fix
* Add clippy workflow ci
* Add Makefile

### CODE REVIEW

File: starky/src/compressor12/compressor12_pil.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ pub enum CompressorPolName {
 impl ToString for CompressorPolName {
     fn to_string(&self) -> String {
         match self {
-            Self::L => format!("L"),
+            Self::L => "L".to_string(),
```
[the remark this answers is not in the document]
Comment (ChengYueJia) [reply starky/src/compressor12/compressor12_pil.rs:26]: I tried to use all of them with format!, however the clippy sugguest to use to_string for "L".

File: plonky/src/api.rs [hunk 2]
```diff
@@ -67,20 +68,22 @@ pub fn prove(
     )?;
 
     let proof = setup.prove(circuit, transcript)?;
-    let writer = std::fs::File::create(proof_bin)?;
+    let writer = std::fs::File::create(proof_bin).expect("Failed to crate proof_bin file");
```
[the remark this answers is not in the document]
Comment (ChengYueJia) [reply plonky/src/api.rs]: Done

File: starky/src/f5g.rs [hunk 3]
```diff
@@ -98,20 +98,18 @@ impl FieldExtension for F5G {
...
         if self.dim == rhs.dim {
             self.cube == rhs.cube
+        } else if self.dim == 1 {
+            self.cube[0] == rhs.cube[0]
+                && rhs.cube[1] == Fr::ZERO
+                && rhs.cube[2] == Fr::ZERO
+                && rhs.cube[3] == Fr::ZERO
+                && rhs.cube[4] == Fr::ZERO
         } else {
-            if self.dim == 1 {
```
[the remark this answers is not in the document]
Comment (ChengYueJia) [reply starky/src/f5g.rs:102]: Fow now it works~~

File: starky/src/polsarray.rs [hunk 4]
```diff
@@ -176,8 +168,8 @@ impl PolsArray {
                 )
             };
             n = rs / 8;
-            for l in 0..n {
-                self.array[i][j] = FGL::from(buff[l]);
+            for x in buff {
```
Comment (eigmax) [inline starky/src/polsarray.rs:171]: bug:  buff size may not be n.

File: starky/src/polsarray.rs [hunk 5]
```diff
@@ -98,8 +90,8 @@ impl PolsArray {
             }
         }
 
-        for i in 0..nPols {
```
Comment (eigmax) [inline starky/src/polsarray.rs:101]: defArray size may not be nPols

Comment (ChengYueJia) [reply starky/src/polsarray.rs:101]: Done.

File: starky/src/polsarray.rs (hunk 4 above)
Comment (ChengYueJia) [reply starky/src/polsarray.rs]: Done

**Comment by eigmax** [comment]
LGTM.  @ibmp33 @cyl19970726 @JangoCCC @BC-A @0x1900matrix, plz review too.

**Comment by JangoCCC** [comment]
LGTM. Respect to your work.

**Pull request merged**